client/templates/home/home.js

```diff
@@ -62,11 +62,7 @@ Template.gameList.helpers({
 });
 
 Template.gameList.events({
-    'click .button-small': function(e) {
-        Meteor.call('joinGame', $(e.currentTarget).attr('id'), function(error, results) {
-            if (error) {
-                console.log(error);
-            }
-        });
+    'click #joinGame': function(e) {
+        Router.go("/upload/" + $(e.target).attr("gameId"));
```

Comment:
Now i'm actually not sure if doing the joinGame in the data function on the route is as good. We probably want the route to wait rendering (or whatever) until the joinGame call passes. So, maybe it's better to joinGame outside of the route? or wait through some iron:router hook? what do you guys think?